backends/ent/annotations.go

```diff
@@ -44,17 +46,17 @@ func (backend *Backend) AddAnnotationToDocuments(name, value string, documentIDs
 func (backend *Backend) AddAnnotationToNodes(name, value string, nodeIDs ...string) error {
 	data := ent.Annotations{}
 
-	for _, nodeID := range nodeIDs {
-		result, err := backend.client.Node.Query().
-			Where(node.IDEQ(nodeID)).
-			Only(backend.ctx)
-		if err != nil {
-			return fmt.Errorf("querying nodes: %w", err)
-		}
+	nodes, err := backend.client.Node.Query().
+		Where(node.NativeIDIn(nodeIDs...)).
+		All(backend.ctx)
+	if err != nil {
+		return fmt.Errorf("querying Node IDs: %w", err)
+	}
```

Comment:
If no `nodeIDs` are passed in, I believe this query returns no results. I think if none are passed in, it should apply the annotation to all nodes. Also let's not change the error message

```suggestion
	predicates := []predicate.Node{}

	if len(nodeIDs) > 0 {
		predicates = append(predicates, node.NativeIDIn(nodeIDs))
	}

	nodes, err := backend.client.Node.Query().
		Where(predicates...).
		All(backend.ctx)
	if err != nil {
		return fmt.Errorf("querying nodes: %w", err)
	}
```